cpp/src/io/orc/stripe_data.cu

```diff
@@ -1334,7 +1335,10 @@ static __device__ void DecodeRowPositions(orcdec_state_s* s,
          s->top.data.cur_row + s->top.data.nrows < s->top.data.end_row) {
     uint32_t nrows = min(s->top.data.end_row - (s->top.data.cur_row + s->top.data.nrows),
                          min((row_decoder_buffer_size - s->u.rowdec.nz_count) * 2, blockDim.x));
-    if (s->chunk.strm_len[CI_PRESENT] > 0) {
+    // Even though s->chunk.strm_len is zero, there is possibility that there is null mask.
+    // This happens in a struct column with nulls which has child column which doesn't have any
+    // nulls.
+    if (s->chunk.strm_len[CI_PRESENT] > 0 or s->chunk.valid_map_base) {
```

Comment:
is it sufficient to check `valid_map_base`?

Reply:
Yeah, else it will be nullptr